docs/production/deployment.md

```diff
@@ -378,3 +378,182 @@ that can be quite difficult to debug.  Be sure to declare your
 unexpectedly (e.g. pointing to a DNS name that you haven't configured
 with multiple IPs for your Zulip machine; sometimes this happens with
 IPv6 configuration).
+
+
+## System and deployment configuration
+
+The file `/etc/zulip/zulip.conf` is used to configure properties of
+the system and deployment; `/etc/zulip/settings.py` is used to
+configure the application itself.  The `zulip.conf` sections and
+settings are described below.
+
+### `[machine]`
+
+#### `puppet_classes`
+
+A comma-separated list of the Puppet classes to install on the server.
+The most common is **`zulip::profile::standalone`**, used for a
+stand-alone single-host deployment.
+[Components](../overview/architecture-overview.html#components) of
+that include:
+ - **`zulip::profile::app_frontend`**
+ - **`zulip::profile::memcached`**
+ - **`zulip::profile::postgresql`**
+ - **`zulip::profile::redis`**
+ - **`zulip::profile::rabbitmq`**
+
+If you are using a [Apache as a single-sign-on
+authenticator](../production/authentication-methods.html#apache-based-sso-with-remote-user),
+you will need to add **`zulip::apache_sso`** to the list.
+
+#### `pgroonga`
+
+Set to the string `enabled` if enabling the [multi-language PGroonga
+search
+extension](../subsystems/full-text-search.html#an-optional-full-text-search-implementation).
+
+
+
+### `[deployment]`
+
+#### `deploy_options`
+
+Options passed by `upgrade-zulip` and `upgrade-zulip-from-git` into
+`upgrade-zulip-stage-2`.  These might be any of:
+
+ - **`--skip-puppet`** skips doing Puppet/apt upgrades.  The user will need
+   to run `zulip-puppet-apply` manually after the upgrade.
+ - **`--skip-migrations`** skips running database migrations.  The
+   user will need to run `./manage.py migrate` manually after the upgrade.
+ - **`--skip-purge-old-deployments`** skips purging old deployments;
+   without it, only deployments with the last two weeks are kept.
+
+Generally installations will not want to set any of these options; the
+`--skip-*` options are primarily useful for reducing upgrade downtime
+for servers that are upgraded frequently by core Zulip developers.
```

Comment:
Looks good; 👍 to making explicit what it means that users will need to manually do.

I'll note that this explicitly does not include `--ignore-static-assets` and `--audit-fts-indexes` flags, since those don't make sense to be applied on every deployment.

Reply:
Yeah, that sounds correct to me.  The `--help` seems like the only place those should be documented (along with the specific docs for when you need them, which is already true).